src/attackmate/variablestore.py

```diff
@@ -22,20 +74,33 @@ def remove_sign(self, name: str, sign: str = '$') -> str:
             return name
 
     def substitute_str(self, template_str: str, blank: bool = False) -> str:
-        temp = Template(template_str)
+        temp = ListTemplate(template_str)
         if blank:
             try:
-                return temp.substitute(self.variables)
+                return temp.substitute(self.variables | self.get_lists_variables())
             except KeyError:
                 return ''
         else:
-            return temp.safe_substitute(self.variables)
+            return temp.safe_substitute(self.variables | self.get_lists_variables())
 
-    def set_variable(self, variable: str, value: str):
-        self.variables[self.remove_sign(variable)] = value
+    def set_variable(self, variable: str, value: str | list[str]):
+        if isinstance(variable, str):
+            varname = self.remove_sign(variable)
+            if isinstance(value, str):
+                if self.is_list(varname):
+                    parsed = self.parse_list(varname)
```

Comment:
for easier understanding unpack already here: 
list_name, index = self.parse_list(varname)
self.lists[list_name][index] = value